regions-ui/tests/app.spec.ts

```diff
@@ -934,3 +576,174 @@ test('region page default info', async ({ page }, testInfo) => {
+    const seenUrls = (testInfo as ExtendedTestInfo).seenUrls;
+
+    await page.goto('http://localhost:5173/region?id=21654846#layer=States+and+territories&region=AUSTRALIAN+CAPITAL+TERRITORY');
+
+    // Wait for images to load
+    await page.waitForLoadState('networkidle');
+
+    // Verify that 3 layers are now visible
+    const leafletLayers = page.locator('div.leaflet-layer');
+    expect(await leafletLayers.count()).toEqual(3); // base layer, this area, species points
+
+    // Verify area name is visible
+    const breadcrumb = page.locator('li', {
+        hasText: 'AUSTRALIAN CAPITAL TERRITORY',
+    });
+    await expect(breadcrumb).toBeVisible(); // check the breadcrumb is visible
+    const h2 = page.locator('h2', { hasText: 'AUSTRALIAN CAPITAL TERRITORY' });
+    await expect(h2).toBeVisible(); // check the button is visible
+
+    // Verify counts
+    const occurrenceHeading = page.getByRole('heading', {
+        name: /Occurrence records \(\d+(\.\d+)?M?\)/,
+    });
+    await expect(occurrenceHeading).toBeVisible();
+    // const occurrenceCount = page.locator('h3', {
+    //     hasText: 'Occurrence records (4.31M)',
+    // }); // count from speciesGroups.json
+    // await expect(occurrenceCount).toBeVisible(); // check the count is visible
+    const speciesCount = page.getByRole('heading', {
+        name: /Number of species \(\d+(,\d+)?\)/,
+    });
+    await expect(speciesCount).toBeVisible(); // check the count is visible
+
+      // Verify some species
+    const speciesAF = page.locator('div[class^="_speciesName_"]', {
+        hasText: 'Aaaaba fossicollis',
+    });
+    const species2 = page.locator('div[class^="_speciesName_"]', {
+        hasText: 'Abantiades labyrinthicus',
+    });
+    await expect(speciesAF).toBeVisible(); // check the species is visible
+    await expect(species2).toBeVisible(); // check the species is visible
+
+    const speciesAFCount = speciesAF.locator('xpath=following-sibling::div[@data-testid="speciesCount"]');
+    var speciesCountAFText = await speciesAFCount.textContent();
+    await speciesCountAFText === '16' ; // check the species count is correct
+
+    await speciesAF.click();
+    const speciesProfileButton = page.locator('button', { hasText: 'Species profile' });
+    await expect(speciesProfileButton).toBeVisible();
+
+    const listRecordsButton = page.locator('button', { hasText: 'List records' });
+    await expect(listRecordsButton).toBeVisible();
+
+
+    // Does not work with mock data
+    // const [biePage] = await Promise.all([
+    //     page.waitForEvent('popup'),
+    //     speciesProfileButton.click(),
+    // ]);
+    //
+    // await expect(biePage).not.toBeNull();
+    // await expect(biePage).toHaveURL(/species\/https:\/\/biodiversity\.org\.au\/afd\/taxa\/[0-9a-fA-F\-]{36}/);
+
+    const [biocachePage] = await Promise.all([
+        page.waitForEvent('popup'),
+        listRecordsButton.click(),
+    ]);
+    await expect(biocachePage).not.toBeNull();
```

Comment:
The destination page needs a mock

Reply:
I will be changing how the button operates. It will end up as an <a> with a href that can be compared to the exact value. If not mocking external website, comment out for now.

Reply:
Params in url are verified now